kalite/contentload/management/commands/channels/khan.py

```diff
@@ -79,39 +90,100 @@ def build_full_cache(items, id_key="id"):
     """
     for item in items:
```

Comment:
Also, it's a triple for loop. Good candidate for porting to C @66eli77 @rjzheng.